Compression/src/CompressDigiMCs_module.cc

```diff
@@ -429,6 +434,31 @@ void mu2e::CompressDigiMCs::produce(art::Event & event)
       throw cet::exception("CompressDigiMCs") << "The number of CrvDigiMCs before and after compression does not match ("
                                               << crvDigiMCs.size() << " != " << _newCrvDigiMCs->size() << ")" << std::endl;
     }
+
```

Comment:
My understanding of the use case for saving these  CrvSteps was for when they are generated by the primary particle, to study Crv pathologies.  However I don't see any logic here to select steps from the primary (or its daughters).  Isn't that necessary?

Reply:
Hi Dave, so there's no need for any logic here because we are just identifying whole CrvStepCollections that we don't want compressed.

Up to this point in the code, we are writing out CrvSteps from all sources (primary, pileup, etc.) that have got into a CrvDigiMC. Here then, we go through any CrvStepCollections that we still want to keep, check if that CrvStep has already been seen, and if it hasn't been seen, we add it to the output collection. 

For the CRV studies, they want all the CrvSteps from the primary stage and so we set _crvStepsToKeep to  ["CosmicS1"] in epilog_cosmic.fcl in Mu2e/Production#277). For all other datasets, we can leave the _crvStepsToKeep array empty.

I hope that makes sense.